src/test/java/com/example/joy/myFirstSpringBoot/controllers/BirthdayInfoControllerIT.java

```diff
@@ -0,0 +1,110 @@
+package com.example.joy.myFirstSpringBoot.controllers;
+
+import com.example.joy.myFirstSpringBoot.services.BasicBirthdayService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.user;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ContextConfiguration(classes = { BirthdayInfoController.class,
+								BasicBirthdayService.class})
+@AutoConfigureMockMvc
+@WebMvcTest
+class BirthdayInfoControllerIT {
+
+	private final static String TEST_USER_ID = "user-id-123";
+	String bd1 = LocalDate.of(1979,7,14).format(DateTimeFormatter.ISO_DATE);
+	String bd2 = LocalDate.of(2018,1,23).format(DateTimeFormatter.ISO_DATE);
+	String bd3 = LocalDate.of(1972, 3, 17).format(DateTimeFormatter.ISO_DATE);
+	String bd4 = LocalDate.of(1945, 12, 2).format(DateTimeFormatter.ISO_DATE);
+	String bd5 = LocalDate.of(2003, 8, 4).format(DateTimeFormatter.ISO_DATE);
+
+	@Autowired
+	private  MockMvc mockMvc;
+	
+	@Test
+	public void testGetBirthdayDOW() throws Exception {
+		testDOW(bd1,"SATURDAY");
+		testDOW(bd2,"TUESDAY");
+		testDOW(bd3,"FRIDAY");
+		testDOW(bd4,"SUNDAY");
+		testDOW(bd5,"MONDAY");
+	}
+
+	@Test
+	public void testGetBirthdayChineseSign() throws Exception {
+		testZodiak(bd1,"Sheep");
+		testZodiak(bd2,"Dog");
+		testZodiak(bd3,"Rat");
+		testZodiak(bd4,"Rooster");
+		testZodiak(bd5,"Sheep");
+	}
+	
+	@Test
+	public void testGetBirthdaytestStarSign() throws Exception {
+		testStarSign(bd1,"Cancer");
+		testStarSign(bd2,"Aquarius");
+		testStarSign(bd3,"Pisces");
+		testStarSign(bd4,"Sagittarius");
+		testStarSign(bd5,"Leo");
+	}
+
+	private void testDOW(String birthday, String dow) throws Exception {
+		MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/birthday/dayOfWeek")
+					.with(user(TEST_USER_ID))
```

Comment:
using `user()` was working more consistently, the first `test*` call would work then the next would fail.
Setting the user on each request fixed it.  Feel free to dig into this more if you like the annotation functionality better